lib/octokit/client/source_import.rb

```diff
@@ -0,0 +1,91 @@
+      # @option options [String] :vcs_password If authentication is required, the password to provide to vcs_url.
+      # @option options [String] :tfvc_project For a tfvc import, the name of the project that is being imported.
+      # @return [Sawyer::Resource] Hash representing the repository import
+      # @see https://developer.github.com/v3/migration/source_imports/#start-an-import
+      #
+      # @example
+      #   @client.import("octokit/octokit.rb", "subversion", "http://svn.mycompany.com/svn/myproject" {
+      #    :vcs_username" => "octocat",
+      #    :vcs_password  => "secret"
+      #   })
+      def start_source_import(repo, vcs, vcs_url, options = {})
+        options = ensure_api_media_type(:source_imports, options.merge(:vcs => vcs, :vcs_url => vcs_url))
+        put "#{Repository.path repo}/import", options
+      end
+
+      # View the progress of an import.
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
+      # @return [Sawyer::Resource] Hash representing the progress of the import
+      # @see https://developer.github.com/v3/migration/source_imports/#get-import-progress
+      #
+      # @example
+      #   @client.source_import_progress("octokit/octokit.rb")
+      def source_import_progress(repo, options = {})
+        options = ensure_api_media_type(:source_imports, options)
+        get "#{Repository.path repo}/import", options
+      end
+
+      # List source import commit authors
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
+      # @return [Array<Sawyer::Resource>] Array of hashes representing commit_authors.
+      # @see https://developer.github.com/v3/migration/source_imports/#get-commit-authors
+      #
+      # @example
+      #   @client.source_import_commit_authors("octokit/octokit.rb")
+      def source_import_commit_authors(repo, options = {})
+        options = ensure_api_media_type(:source_imports, options)
+        get "#{Repository.path repo}/import/authors", options
+      end
+
+      # Update an author's identity for the import.
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
+      # @param author [Integer] The id of the commit author
+      # @param values [Hash] The updated author attributes
+      # @option values [String] :email The new Git author email.
+      # @option values [String] :name The new Git author name.
+      # @return [Sawyer::Resource] Hash representing the updated commit author
+      # @see https://developer.github.com/v3/migration/source_imports/#map-a-commit-author
+      #
+      # @example
+      #   @client.map_source_import_commit_author("octokit/octokit.rb", 1, {
+      #     :email => "hubot@github.com",
+      #     :name => "Hubot the Robot"
+      #   })
+      def map_source_import_commit_author(repo, author, values, options = {})
```

Comment:
Instead of building this with `repo` and the pretty esoteric `author_id` what if we leaned into hypermedia and have this method take a `url`. That you get from `source_import_commit_authors`? I'm guessing that's the only way you'd know about those authors anyway.